aerogear-android-push/src/main/java/org/jboss/aerogear/android/unifiedpush/gcm/GCMSharedPreferenceProvider.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * JBoss, Home of Professional Open Source
+ * Copyright Red Hat, Inc., and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.aerogear.android.unifiedpush.gcm;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import org.jboss.aerogear.android.core.Provider;
+
+/**
+ * Provides a standard shared preferences reference
```

Comment:
If we receive the update registrationID message while the application is offline we need to be able to load the registrar associated with that ID.  Right now the easiest thing to do is to store the configuration associated with the ID when .register is called.